### PR TITLE
docs: fix broken links and references to tedge-configuration-plugin

### PR DESCRIPTION
**Pull request by reubenmiller**
## Proposed changes

Update documentation to address the following:

* Fix broken links (introduced in https://github.com/thin-edge/thin-edge.io/pull/2380 and https://github.com/thin-edge/thin-edge.io/pull/2235)
* Replace references to the deprecated c8y-configuration-plugin (as the tedge-configuration-plugin was mentioned in some of the broken links)

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/src/references/agent/restart-operation.md [hunk 1]
```diff
@@ -8,20 +8,20 @@ sidebar_position: 4
 
 Thin-edge defines a `restart` operation to restart a device, being the main device or a child device.
 
-- A restart is typically triggered by a [mapper](../mappers) on behalf of a cloud operator.
+- A restart is typically triggered by a [mapper](../mappers/index.md) on behalf of a cloud operator.
```
Comment (didier-wenzek) [inline docs/src/references/agent/restart-operation.md:11]: Is this mandatory or a best practice?

**Review by didier-wenzek — APPROVED** [review]
Thank you for fixing these broken links I badly introduced.

File: docs/src/references/agent/restart-operation.md (hunk 1 above)
Comment (reubenmiller) [reply docs/src/references/agent/restart-operation.md:11]: I think it is best practice, but not 100% sure. Though it is inline with all of the references in the docs (all seem to use `/index.md`).

**Comment by reubenmiller** [comment]
I've had to skip a system test due to system test which sometimes passes when it shouldn't #2383

**Pull request merged**